docs: describe composition of selectors in pipeable operators

dherges opened the pull request:
Ready for review.

Closes #1002

MattiJarvinen-BA commented on docs/store/selectors.md:
```diff
@@ -213,3 +213,61 @@ class MyAppComponent {
+import { map, filter } from 'rxjs/operators';
+
+store.pipe(
+  select(selectProjectedValues(state),
```
Missing one right parenthesis.

MattiJarvinen-BA commented on docs/store/selectors.md:
````diff
@@ -213,3 +213,61 @@ class MyAppComponent {
+store.pipe(selectProjectedValuesFiltered)
+     .subcribe(/* .. */);
+```
+
````
Hmm... this is an interesting one although I might add a new pipeable operator that just leaves out undefined values. It would give more reusability throughout the application.

dherges commented:
@brandonroberts Here's my write-up on combining selectors and pipeable operators. I also added a stackblitz to demosntrate the advanced example of such a read model. Maybe the comments in the code lkisting could be better? Wdyt?

/FYI @dummdidumm @MattiJarvinen-BA

brandonroberts commented:
@dherges I think the writeup is great! I also feel like it should be a separate blog post in itself and it should be under an `Advanced Usage` section in the store docs. Will you add a header above your section for `Advanced Usage`?

dherges commented:
I updated, removed some typos, and added the "Advanced Usage" headline. I think the basics how the selector compares to the rxjs pipeable operators should be defo in the docs - that's what I was looking for initially. I'd either take the advanced example for a blog post or blog something on the ngrx/cqrs analogies. The server-side/client-side filtering is a classic example and worth a good blog post.

brandonroberts commented:
Thanks!
